helm/exporter-kube-state/templates/kube-state-metrics.rules.yaml

```diff
@@ -3,7 +3,8 @@ groups:
 - name: kube-state-metrics.rules
   rules:
   - alert: DeploymentGenerationMismatch
-    expr: kube_deployment_status_observed_generation != kube_deployment_metadata_generation
```

Comment:
Rather than changing this file make the changes [here](https://github.com/coreos/prometheus-operator/tree/master/contrib/kube-prometheus/assets/prometheus/rules) and run `helm/hack/sync_kube_prometheus.py`

Reply:
@brancz despite the change in the helm template, are you ok with the changes in kube-prometheus alerts?

Reply:
I'm struggling to understand what this gets us at alert time, the alert already says the namespace and deployment this belongs to, how do the additional labels help?

Reply:
An example from our inhouse use is to add a `team` label to the deployments and pods. This is used to route the alert to the correct group in PagerDuty.
This part could also have been done with the use of ServiceMonitor as you (@brancz) suggested in https://github.com/coreos/prometheus-operator/issues/695#issuecomment-339652103. That is not implemented yet I take?

Also labels like `tier`, and other infrastructural mapping labels can be used in alerts to inhibit and group alerts.  

I would love having a simpler way of doing this, and having it upstream until a better way is all I suggest in this PR.

Reply:
I think what you're trying to achieve is absolutely in line with what we want to enable with the Prometheus Operator, though I'd prefer if we just get https://github.com/coreos/prometheus-operator/issues/695#issuecomment-339652103 down :slightly_smiling_face: .

Reply:
Alright. I'll let this PR lie around for doc purposes if someone wants to achieve the same while waiting for #695

Reply:
I opened #1145 to get this done :slightly_smiling_face: